go/exocom-mock/exocom-mock.go

```diff
@@ -28,42 +30,62 @@ func New() *ExoComMock {
 }
 
 // Close takes this ExoComMock instance offline
-func (exoCom *ExoComMock) Close() error {
-	return exoCom.server.Close()
+func (e *ExoComMock) Close() error {
+	return e.server.Close()
 }
 
 // Listen brings this ExoComMock instance online
-func (exoCom *ExoComMock) Listen(port int) error {
-	exoCom.server.Addr = fmt.Sprintf(":%d", port)
-	return exoCom.server.ListenAndServe()
+func (e *ExoComMock) Listen(port int) error {
+	e.server.Addr = fmt.Sprintf(":%d", port)
+	return e.server.ListenAndServe()
 }
 
 // HasConnection returns whether or not a socket is connected
-func (exoCom *ExoComMock) HasConnection() bool {
-	return exoCom.socket != nil
+func (e *ExoComMock) HasConnection() bool {
+	return e.socket != nil
 }
 
-// Reset nils the socket and clears all received messages
-func (exoCom *ExoComMock) Reset() {
-	exoCom.socket = nil
-	exoCom.ReceivedMessages = []structs.Message{}
+// Reset closes and nils the socket and clears all received messages
+func (e *ExoComMock) Reset() {
+	if e.HasConnection() {
+		err := e.socket.Close()
+		if err != nil {
```

Comment:
I like that idea when the function call is short but we have many other similar instances where the function call is long. For consistency, I'd prefer to leave it like this